src/Todo/Model.hs

```diff
@@ -32,3 +42,58 @@ data Todo = Todo
     , todoDeadline    :: Maybe Deadline
     , todoDescription :: Maybe String
     } deriving (Eq, Show)
+
+data TodoDB = TodoDB 
+    { tododbTodos :: Todos
+    , tododbTags  :: [Tag]
+    } deriving (Eq, Show)
+
+
+addTodo :: Todo -> TodoDB -> TodoDB
+addTodo todo TodoDB { .. } = TodoDB { tododbTodos = todo : tododbTodos, .. }
+
+addTag :: Tag -> TodoDB -> TodoDB
+addTag tag TodoDB { .. } = TodoDB { tododbTags = tag : tododbTags, .. }
+
+deleteTag :: Tag -> TodoDB -> TodoDB
+deleteTag tag db@TodoDB { .. } = case List.find ((tag :| [] ==) . todoTags) tododbTodos of 
+    Just _  -> db
+    Nothing -> TodoDB { tododbTodos = map (\Todo{ .. } -> 
+        Todo { todoTags = fromList $ List.delete tag . toList $ todoTags, .. }) tododbTodos
+                      , tododbTags = List.delete tag tododbTags }
+
+editTag :: Tag -> Tag -> TodoDB -> TodoDB
+editTag old new TodoDB{ .. } = 
+    TodoDB { tododbTodos = map (\Todo{ .. } -> 
+        Todo { todoTags = fromList $ replace $ toList todoTags, .. }) tododbTodos
+           , tododbTags = new : List.delete old tododbTags } 
+  where 
+    replace :: [Tag] -> [Tag]
+    replace []      = []
+    replace (t:ts) 
+        | t == old  = new : ts 
+        | otherwise = t : replace ts
+
+
+defTodo :: Title -> Priority -> Tags -> Todo
+defTodo todoTitle todoPriority todoTags = Todo { todoState = TODO
+                                               , todoSubtasks = []
+                                               , todoDeadline = Nothing
+                                               , todoDescription = Nothing
+                                               , .. }
+
+getByState :: State -> Todos -> Todos
+getByState state = filter $ (state ==) . todoState
+
+getByTag :: Tag -> Todos -> Todos
+getByTag tag = filter $ elem tag . todoTags
+
+getTopPrio :: Int -> Todos -> Todos
+getTopPrio n lst = take n (List.sortBy ((. todoPriority) . compare . todoPriority) lst)
```

Comment:
`(. todoPriority)` is not a good and readable code... Use [`comparing`](https://hackage.haskell.org/package/base-4.11.0.0/docs/Data-Ord.html#v:comparing) instead. Btw, I still think that it's better to implement logic _the bigger value of `Priority` is then the most important the task_. Because now if you have task with priority `0` you can't add more important task without reassigning priorities to the rest task.